gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -231,9 +235,23 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
       mapContainerFragment = restoreChildFragment(savedInstanceState, MapContainerFragment.class);
     }
 
+    updateNavHeader();
     setUpBottomSheetBehavior();
   }
 
+  private void updateNavHeader() {
+    View navHeader = binding.navView.getHeaderView(0);
+    User user = authenticationManager.getCurrentUser();
+
+    TextView userDisplayNameTv = navHeader.findViewById(R.id.user_display_name);
+    TextView userEmailTv = navHeader.findViewById(R.id.user_email);
+    ImageView userPhotoIv = navHeader.findViewById(R.id.user_image);
+
+    userDisplayNameTv.setText(user.getDisplayName());
+    userEmailTv.setText(user.getEmail());
+    Picasso.get().load(user.getPhotoUrl()).placeholder(R.drawable.ground_logo).into(userPhotoIv);
```

Comment:
This unfortunately won't work when offline.. Maybe add a TODO with a bug to cache the pic offline?

Reply:
Picasso already does the caching for us